cpp/src/parquet/geospatial_statistics.h

```diff
@@ -0,0 +1,192 @@
+  /// (such that is_set() returns false and they should not be written).
+  EncodedGeoStatistics Encode() const;
+
+  /// \brief Returns true if all WKB encountered was valid or false otherwise
+  bool is_valid() const;
+
+  /// \brief Reset existing statistics and populate them from previously-encoded ones
+  void Decode(const EncodedGeoStatistics& encoded);
+
+  /// \brief The minimum encountered value in the X dimension, or Inf if no non-NaN X
+  /// values were encountered.
+  ///
+  /// The Parquet definition allows for "wrap around" bounds where xmin > xmax. In this
+  /// case, these bounds represent the union of the intervals [xmax, Inf] and [-Inf,
+  /// xmin]. This implementation does not yet generate these types of bounds but they may
+  /// be encountered in files written by other writers.
+  double xmin() const;
+
+  /// \brief The maximum encountered value in the X dimension, or -Inf if no non-NaN X
+  /// values were encountered, subject to "wrap around" bounds (see xmin()).
+  double xmax() const;
+
+  /// \brief The minimum encountered value in the Y dimension, or Inf if no non-NaN Y
+  /// values were encountered. Wrap around bounds are not permitted in the Y dimension.
+  double ymin() const;
+
+  /// \brief The maximum encountered value in the Y dimension, or -Inf if no non-NaN Y
+  /// values were encountered. Wrap around bounds are not permitted in the Y dimension.
+  double ymax() const;
+
+  /// \brief The minimum encountered value in the Z dimension, or Inf if no non-NaN Z
+  /// values were encountered. Wrap around bounds are not permitted in the Z dimension.
+  double zmin() const;
+
+  /// \brief The maximum encountered value in the Z dimension, or -Inf if no non-NaN Z
+  /// values were encountered. Wrap around bounds are not permitted in the Z dimension.
+  double zmax() const;
+
+  /// \brief The minimum encountered value in the M dimension, or Inf if no non-NaN M
+  /// values were encountered.  Wrap around bounds are not permitted in the M dimension.
+  double mmin() const;
+
+  /// \brief The maximum encountered value in the M dimension, or -Inf if no non-NaN M
+  /// values were encountered.  Wrap around bounds are not permitted in the M dimension.
+  double mmax() const;
+
+  /// \brief All minimum values in XYZM order
+  std::array<double, 4> lower_bound() const;
+
+  /// \brief All maximum values in XYZM order
+  std::array<double, 4> upper_bound() const;
+
+  /// \brief Returns true if zero finite coordinates or geometry types were encountered
+  ///
+  /// This will occur, for example, if all encountered values were null. If all
+  /// encountered values were EMPTY (e.g., all values were POINT EMPTY, LINESTRING EMPTY,
+  /// etc.), is_empty() will still return false because the geometry types list will
+  /// contain values.
+  bool is_empty() const;
+
+  /// \brief Returns true if any non-NaN X values were encountered or false otherwise
+  bool has_x() const;
```

Comment:
Do we also want a convenience method to return all 4 predicates? Something like:
```c++
  /// \brief Return whether each of X, Y, Z, M coordinate is present
  std::array<bool, 4> has_coordinates() const;
```

Reply:
(it can also make the tests more concise :))